src/models/connectionCredentials.ts

```diff
@@ -281,6 +281,14 @@ export class ConnectionCredentials implements IConnectionCredentials {
         return authenticationType === utils.authTypeToString(AuthenticationTypes.SqlLogin);
     }
 
+    public static isPasswordBasedConnectionString(connectionString: string): boolean {
+        return connectionString.toLowerCase().includes('password');
+    }
+
+    public static isConnectionStringSaved(connectionString: string): boolean {
+        return connectionString.includes(ConnectionStore.CRED_CONNECTION_STRING_PREFIX);
```

Comment:
Would rather we use a separate field for this (like you do with the password). The phrase "connectionString:" could end up in a normal connection string (such as if a database contained that string). And given our discussion of the security concerns I would say it's worth making this more robust for that case.